bentoml/_internal/runner/runner.py

```diff
@@ -183,238 +140,179 @@ def get_runnable_methods(cls) -> t.List[RunnableMethod]:
 
-    @property
-    def models(self):
-        return self._models
+        self._runnable = runner.runnable_class()
 
-    @property
-    def resource_config(self):
-        # look up configuration
-        # else return self._resource_config
-        ...
+    def destroy_local(self):
+        if not self._runnable:
+            logger.warning("local runnable not found")
+        else:
+            del self._runnable
 
-    def _run(self, runner_method_name, *args, **kwargs):
-        if self._runner_app_client:
-            return self._runner_app_client.run(runner_method_name, *args, **kwargs)
+    def init_client(self, runner: Runner):
+        if self._runner_client:
+            logger.warning("re creating remote runner client")
         if self._runnable:
-            return self._runnable[runner_method_name](*args, **kwargs)
+            raise BentoMLException("TODO: ..")
 
+        self._runner_client = RemoteRunnerClient(runner)
+
+    def destroy_client(self):
+        if not self._runner_client:
+            logger.warning("remote runner client not found")
+        else:
+            del self._runner_client
+
+    def run_method(
+        self,
+        method_name: str,
+        *args: t.Any,
+        **kwargs: t.Any,
+    ) -> t.Any:
+        if self._runnable is not None:
+            return getattr(self._runnable, method_name)(*args, **kwargs)
+        if self._runner_client is not None:
+            return self._runner_client.run_method(method_name, *args, **kwargs)
         raise BentoMLException(
             "runner not initialized"
         )  # TODO: make this UninitializedRunnerException
 
-    async def _async_run(self, runner_method_name, *args, **kwargs):
-        if self._runner_app_client:
-            return await self._runner_app_client.async_run(
-                runner_method_name, *args, **kwargs
-            )
-        if self._runnable:
+    async def async_run_method(
+        self,
+        method_name: str,
+        *args: t.Any,
+        **kwargs: t.Any,
+    ) -> t.Any:
+        if self._runnable is not None:
             import anyio
 
-            # TODO(jiang): to_thread
-            # return await self._runnable[runner_method_name](*args, **kwargs)
-
+            method = getattr(self._runnable, method_name)
+            return anyio.to_thread.run_sync(method, *args, **kwargs)
+        if self._runner_client is not None:
+            return await self._runner_client.async_run_method(
+                method_name,
+                *args,
+                **kwargs,
+            )
         raise BentoMLException(
             "runner not initialized"
         )  # TODO: make this UninitializedRunnerException
 
+
+@attr.define(frozen=True)
+class Runner:
+    runnable_class: t.Type[Runnable] = attr.field()
+    init_params: t.Dict[str, t.Any] = attr.field()
+    name: str = attr.field()
+    strategy: t.Type[Strategy] = attr.field(default=DefaultStrategy)
```

Comment:
Maybe either `schedule_strategy` or `scheduling_strategy`. Strategy is maybe too generic.